File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
  def maxProfit(self, prices: List[int]) -> int:
    min_price = float("inf")
    max_profit = 0

    for i in range(1, len(prices)):
      min_price = min(min_price, prices[i - 1])
      max_profit = max(max_profit, prices[i] - min_price)

    return max_profit

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
  let min = Infinity;
  let maxProfit = -Infinity;

  for (let i = 0; i < prices.length - 1; i++) {
    min = Math.min(min, prices[i]);
    maxProfit = Math.max(prices[i + 1] - min, maxProfit);
  }

  return maxProfit > 0 ? maxProfit : 0
}
